client.go

```diff
@@ -87,50 +88,92 @@ func newBufferPool() *sync.Pool {
 
 // Count returns nothing and allows a counter to be incremented by a value
 func (c *Client) Count(name string, value int) {
-
-	c.send(name, value, "c") // for a counter
-
+	c.send(name, value, "c", "sum") // for a counter
 }
 
 // Timer returns nothing and allows a timer metric to be set
 func (c *Client) Timer(name string, value int) {
+	c.send(name, value, "ms", "sum") // timer, so count in milliseconds
+}
 
-	c.send(name, value, "ms") // timer, so count in milliseconds
-
+// AverageTimer returns nothing and allows a timer metric to be set
+func (c *Client) AverageTimer(name string, value int) {
+	c.send(name, value, "ms", "avg") // timer, so count in milliseconds
 }
 
 // Send is used to record a metric and have it send to
 // the bucky server - this is thread safe
-func (c *Client) send(name string, value int, unit string) {
-
+func (c *Client) send(name string, value int, unit string, action string) {
 	// Send to channel so we never block
-	c.input <- MetricWithValue{name, value, unit}
+	m := Metric{
+		name: name,
+		unit: unit,
+	}
+
+	v := Value{}
+
+	switch action {
+	case "sum":
+		v.Sum = &Sum{
+			Value: value,
+		}
+	case "avg":
+		v.Avg = &Average{}
+
+		var avgResult int
+		var newCount int
+
+		if val, ok := c.metrics[m]; ok {
```

Comment:
I merged this thinking i'd reviewed it on friday and it was ok - but I suspect i'm lying to myself - apologies :)